src/common/AdaptiveStream.cpp

```diff
@@ -909,8 +909,26 @@ bool AdaptiveStream::ensureSegment()
       std::this_thread::sleep_for(std::chrono::milliseconds(100));
       return false;
     }
+    else if (tree_.HasUpdateThread() && current_period_ != tree_.periods_.back())
+    {
+      auto itPer = std::find(tree_.periods_.begin(), tree_.periods_.end(), current_period_);
+      if (itPer == tree_.periods_.end())
+      {
+        LOG::LogF(LOGWARNING, "Current Period missing from tree!");
+        state_ = STOPPED;
+        return false;
+      }
+
+      auto pos = std::distance(tree_.periods_.begin(), itPer);
+      auto newPd = tree_.periods_[pos+1];
```

Comment:
its safe `pos+1`? no need to check for possible out of range?

Reply:
Yes because the range check happens in the two enclosing conditionals. We know it's not at the back, and we know its in the range.